core/src/types/read/buffer_stream.rs

```diff
@@ -125,27 +125,28 @@ impl oio::Read for ChunkedReader {
     }
 }
 
-/// BufferStream is a stream of buffers.
+/// BufferStream is a stream of buffers, created by [`Reader::into_stream`]
 ///
+/// `BufferStream` implements `Stream` trait.
+pub struct BufferStream {
+    state: State,
+}
+
 /// # Notes
```

Comment:
This comment would now go to `State`'s? Looks weird.

Reply:
> This comment would now go to `State`'s? Looks weird.

Yep, I don't want users (from public API) to read this.

Reply:
But it's not for `State`? Maybe `//` on BufferStream.